tests/hierarchical/hierarchical_non_uniform_local_range.cpp

```diff
@@ -92,9 +92,12 @@ template <int dim> void check_dim(util::logger &log) {
                 local_range_total>(local_range, local_range);
         cgh.parallel_for_work_group<kernel<dim>>(
             groupRange, localRange, [=](sycl::group<dim> group_pid) {
+              sycl::range<dim> r;
+              auto group_id = group_pid.get_group_id();
+              for (int i = 0; i < dim; ++i) r[i] = group_id[i];
```

Comment:
On the other hand having to write this ugly code means we missed our goals somewhere.
Perhaps for SYCL Next exposing a range interface for `id` and an *explicit* constructor from a range to a `range` would make it better?